boxoffice/views/order.py

```diff
@@ -433,6 +435,52 @@ def cancel_line_item(line_item):
     return make_response(jsonify(status='ok', result={'message': 'Ticket cancelled', 'cancelled_at': date_format(line_item.cancelled_at)}), 200)
 
 
+def process_partial_refund_for_order(order, form_dict):
+    form = OrderRefundForm.from_json(form_dict, meta={'csrf': False})
+    if form.validate():
+        requested_refund_amount = form.amount.data
+        if not order.paid_amount:
+            return make_response(jsonify(status='error', error='free_order',
+                error_description='Refunds can only be issued for paid orders'), 403)
+
+        if (order.refunded_amount + requested_refund_amount) > order.paid_amount:
+            return make_response(jsonify(status='error', error='excess_partial_refund',
+                error_description='Invalid refund amount, must be lesser than net amount paid for the order'), 403)
+
+        payment = OnlinePayment.query.filter_by(order=order,
+            pg_payment_status=RAZORPAY_PAYMENT_STATUS.CAPTURED).one()
+        rp_resp = razorpay.refund_payment(payment.pg_paymentid, requested_refund_amount)
+        if rp_resp.status_code == 200:
+            transaction = PaymentTransaction(order=order, transaction_type=TRANSACTION_TYPE.REFUND,
+                online_payment=payment, amount=requested_refund_amount, currency=CURRENCY.INR,
+                refunded_at=func.utcnow(), refund_description=form.refund_description.data,
+                internal_note=form.internal_note.data, note_to_user=form.note_to_user.data)
```

Comment:
You should ideally be using `form.populate_obj(transaction)`. Right now you're hardcoding the fields you want, which is splitting logic between the form and the view.